### ISSUE TITLE
bug of not float of a cell 

### ISSUE THREAD

**Issue by jixiuf** [tealeg/xlsx#101]
the cell content is "[1],[12,\"DATE NOT NULL DEFAULT '0000-00-00'\"]"
when I call cell.String() 
I got this error
strconv.ParseFloat: parsing "[1],[12,\"DATE NOT NULL DEFAULT '0000-00-00'\"]":

in getBuiltinNumberFormat()
I found the numFmtId is 49 and it return "@"

**Comment by ShawnMilo** [comment]
Can you post a short, complete program and a sample spreadsheet I can use to duplicate the problem?

**Comment by jixiuf** [comment]
```
package main

import (
	"fmt"
	"github.com/tealeg/xlsx"
)

func main() {
	excelPath, _ := xlsx.OpenFile("/tmp/a.xlsx")
	fmt.Println("cell[0][0].centent", excelPath.Sheets[0].Rows[0].Cells[0].String())
}
```
download /tmp/a.xlsx from https://github.com/jixiuf/hellogit/raw/master/a.xlsx

**Comment by jixiuf** [comment]
sorry ,I click the wrong button

**Comment by ShawnMilo** [comment]
@tealeg: As @jixiuf reported, the `numberFormat` that `func (c *Cell) FormattedValue()` is using is `@`. Also, `c.Type()` is correctly reporting `CellTypeString`. However, numberFormat `@` is catching the switch/case branch `case "0.00", "#,##0.00", "@":`, which attempts to parse it as a float.

I haven't looked at this part of the code before, so I don't know what the format `@` means and whether it's being assigned incorrectly due to another bug, but since, at this point, the Cell is known to be `CellTypeString`, should there just be a check before the `switch` statement that says if it's CellTypeString, just return `c.Value`?

**Comment by ShawnMilo** [comment]
@jixiuf 

This patch fixes the problem for me:

````
diff --git cell.go cell.go
index 46ea1bc..0a38ad7 100644
--- cell.go
+++ cell.go
@@ -193,6 +193,9 @@ func (c *Cell) formatToInt(format string) string {
 
 // Return the formatted version of the value.
 func (c *Cell) FormattedValue() string {
+       if c.Type() == CellTypeString {
+               return c.Value
+       }
        var numberFormat string = c.GetNumberFormat()
        switch numberFormat {
        case "general":
````

@tealeg 

Please let me know if you want me to submit a pull request for this patch, or if there's a more fundamental problem that needs to be fixed.

**Comment by tealeg** [comment]
@ShawnMilo I think what you have above is the correct fix for this issue.

**Comment by ShawnMilo** [comment]
Thanks. It breaks a test, but I suspect the test is wrong. I'll push it up with comments to a new branch.

**Comment by ShawnMilo** [comment]
Note this wasn't actually fixed until `04f14253d2a201985e0969691eca06584a971be4` due to a typo.
